src/companents/modal/companents/add-member.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { StateReduxInterface } from '../../../state';
+import { getUsers, IUser } from '../../../api/requests';
+
+export interface IMember {
+  id: string;
+  login: string;
+  name?: string;
+}
+
+interface IAddMember {
+  setList: React.Dispatch<React.SetStateAction<IMember[]>>;
+  membersId?: string[];
+}
+
+export const AddMember = function (props: IAddMember) {
+  const token = localStorage.getItem('token') as string;
+  const user = useSelector((state: StateReduxInterface) => state.user);
+  const { t } = useTranslation();
+
+  const [members, setMembers] = useState([] as IMember[]);
+  const [current, setCurrent] = useState('');
+  const [usersList, setUserList] = useState([] as IUser[]);
+
+  const [alertError, setAlertError] = useState('');
+  const [preloader, setPreloader] = useState('');
+
+  const add = async () => {
+    if (user.login === current) {
+      setAlertError(t('newMember:alert:you') as string);
+    } else if (!!members.find((e) => e.login === current)) {
+      setAlertError(t('newMember:alert:added') as string);
+    } else {
+      try {
+        setPreloader(' spinner-btn spinner-btn--dark');
+        let users = [...usersList];
+        if (!usersList.length) {
+          users = await getUsers(token);
+          setUserList(users);
+        }
+        const currentUser = users.find((e) => e.login === current);
+        if (!currentUser) {
+          setAlertError(t('newMember:alert:none') as string);
+        } else if (props.membersId?.includes(currentUser._id as string)) {
+          setAlertError(t('newMember:alert:member') as string);
+        } else {
+          const value = {
+            id: currentUser._id as string,
+            login: currentUser.login,
+            name: currentUser.name,
+          };
+          const newUsers = [...members, value];
+          setMembers(newUsers);
+          props.setList(newUsers);
+          setCurrent('');
+        }
```

Comment:
try to keep your logic separated to keep your component code cleaner, otherwise you will easily end up with 1000+ lines long component